pages/en/supported-networks/arweave.mdx

```diff
@@ -0,0 +1,199 @@
+- Arweave data sources introduce an optional source.owner field, which is the public key of an Arweave wallet 
+
+Arweave data sources support two types of handlers:
+
+- `blockHandlers` - Run on every new Arweave block. No source.owner is required.
+- `transactionHandlers` - Run on every transaction where the data source's source.owner is the owner. Note that only exact matches are processed, and if you don't add a source.owner, the subgraph will index every transaction.
+
+> Transactions are the building blocks of the Arweave permaweb and they are objects created by end-users.
+
+> Note: [Bundlr](https://bundlr.network/) transactions are not supported yet.
+
+## Schema Definition
+
+Schema definition describes the structure of the resulting subgraph database and the relationships between entities. This is agnostic of the original data source. There are more details on the subgraph schema definition [here](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#the-graphql-schema).
+
+## AssemblyScript Mappings
+
+The handlers for processing events are written in [AssemblyScript](https://www.assemblyscript.org/).
+
+Arweave indexing introduces Arweave-specific data types to the [AssemblyScript API](https://thegraph.com/docs/en/developer/assemblyscript-api/).
+
+```tsx
+class ProofOfAccess {
+    option: string
+    txPath: Bytes
+    ataPath: Bytes
+    chunk: Bytes
+}
+  
+class Block {
+     timestamp: u64
+     lastRetarget: u64
+     height: u64
+     indepHash: Bytes
+     nonce: Bytes
+     previousBlock: Bytes
+     diff: Bytes
+     hash: Bytes
+     txRoot: Bytes
+     txs: Bytes[]
+     walletList: Bytes
+     rewardAddr: Bytes
+     tags: Tag[]
+     rewardPool: Bytes
+     weaveSize: Bytes
+     blockSize: Bytes
+     cumulativeDiff: Bytes
+     hashListMerkle: Bytes
+     poa: ProofOfAccess
+}
+
+class Transaction {
+     format: u32
+     id: Bytes
+     lastTx: Bytes
+     owner: Bytes
+     tags: Tag[]
+     target: Bytes
+     quantity: Bytes
+     data: Bytes
+     dataSize: Bytes
+     dataRoot: Bytes
+     signature: Bytes
+     reward: Bytes
+}
+```
+
+Writing the mappings of an Arweave Subgraph is very similar to writing the mappings of an Ethereum Subgraph. For more information, click [here](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#writing-mappings).
+
+## Deploying an Arweave Subgraph
+
+Once your subgraph has been created, you can deploy your subgraph by using the `graph deploy` CLI command after running the `graph create` CLI command:
+
+**Hosted Service**
+```bash
+graph create subgraph-name --product hosted-service # creates a subgraph on a local Graph Node (on the Hosted Service, this is done via the UI)
+```
+```bash
+graph deploy --node https://api.thegraph.com/deploy/ --ipfs https://api.thegraph.com/ipfs/ --access-token <your-access-token>
+```
+
+**Local Graph Node (based on default configuration):**
```

Comment:
The problem is that this Firehose endpoint is not "public" and requires authentication access, hence it's not possible to easily give access to a working Firehose Arweave endpoint so that people are able to locally run `graph-node`.

Reply:
Agreed on deleting this section